tests/utils/git_test.py

```diff
@@ -17,63 +17,139 @@
 
 import logging
 import os
+import re
+import shutil
 from unittest import mock
 
 import pytest
 
 from esrally import exceptions
-from esrally.utils import git
+from esrally.utils import git, io, process
 
 
+# pylint: disable=too-many-public-methods
 class TestGit:
+    @classmethod
+    def setup_class(cls):
+        cls.remote_repo = "esrally"
+        cls.local_branch = "rally-unit-test-local-only-branch"
+        cls.remote_branch = "rally-unit-test-remote-only-branch"
+        cls.rebase_branch = "rally-unit-test-rebase-branch"
+
+        # this is assuming that nobody stripped the git repo info in their Rally working copy
+        cls.original_src_dir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+        cls.local_tmp_src_dir = io.escape_path("/tmp/rally-unit-test-local-dir")
+        cls.remote_tmp_src_dir = io.escape_path("/tmp/rally-unit-test-remote-dir")
+        cls.tmp_clone_dir = io.escape_path("/tmp/rally-unit-test-clone-dir")
+
+        # delete any pre-existing dirs (maybe last test setup cancelled early)
+        shutil.rmtree(TestGit.local_tmp_src_dir, ignore_errors=True)
+        shutil.rmtree(TestGit.remote_tmp_src_dir, ignore_errors=True)
+        shutil.rmtree(TestGit.tmp_clone_dir, ignore_errors=True)
```

Comment:
Can you please use `tmp_path` instead of `/tmp`? https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

It makes sure to use the correct /tmp for every OS (macOS currently auto-deletes entries in /tmp after a few days, for example - not an issue here but can be surprising) and means you don't need to care about removing the directories, pytest removes older entries automatically. So you can remove all rmtree calls.

Reply:
Thanks for the pointers! Addressed in https://github.com/elastic/rally/pull/1602/commits/529ea0a8dd8a9cdd47b33890eadbd05ea5cf13e4 and https://github.com/elastic/rally/pull/1602/commits/8f50ce76806a43d42ecf30fc300e7ff48cf724b8